pkg/authorization/defaultrolebindings/project_policy.go

```diff
@@ -23,27 +23,59 @@ const (
 	DeployerServiceAccountName = "deployer"
 )
 
-func GetBootstrapServiceAccountProjectRoleBindings(namespace string) []rbacv1.RoleBinding {
+type projectRoleBindings func(namespace string) []rbacv1.RoleBinding
+type serviceAccountRoleBinding func(namespace string) rbacv1.RoleBinding
+
+func GetBootstrapServiceAccountProjectRoleBindings(namespace string) rbacv1.RoleBinding {
 	imagePuller := newOriginRoleBindingForClusterRoleWithGroup(ImagePullerRoleBindingName, ImagePullerRoleName, namespace, serviceaccount.MakeNamespaceGroupName(namespace))
 	imagePuller.Annotations[openShiftDescription] = "Allows all pods in this namespace to pull images from this namespace.  It is auto-managed by a controller; remove subjects to disable."
 
+	return imagePuller
+}
+
+func GetBuilderServiceAccountProjectRoleBindings(namespace string) rbacv1.RoleBinding {
 	imageBuilder := newOriginRoleBindingForClusterRoleWithSA(ImageBuilderRoleBindingName, ImageBuilderRoleName, namespace, BuilderServiceAccountName)
 	imageBuilder.Annotations[openShiftDescription] = "Allows builds in this namespace to push images to this namespace.  It is auto-managed by a controller; remove subjects to disable."
 
+	return imageBuilder
+}
+
+func GetDeployerServiceAccountProjectRoleBindings(namespace string) rbacv1.RoleBinding {
 	deployer := newOriginRoleBindingForClusterRoleWithSA(DeployerRoleBindingName, DeployerRoleName, namespace, DeployerServiceAccountName)
 	deployer.Annotations[openShiftDescription] = "Allows deploymentconfigs in this namespace to rollout pods in this namespace.  It is auto-managed by a controller; remove subjects to disable."
 
-	return []rbacv1.RoleBinding{
-		imagePuller,
-		imageBuilder,
-		deployer,
+	return deployer
+}
+
+func composeRoleBindings(roleBindings ...serviceAccountRoleBinding) projectRoleBindings {
+	return func(namespace string) []rbacv1.RoleBinding {
+		bindings := []rbacv1.RoleBinding{}
+		for _, rbfunc := range roleBindings {
+			bindings = append(bindings, rbfunc(namespace))
+		}
+		return bindings
+	}
+}
+
+func GetRoleBindingsForController(controller string) projectRoleBindings {
```

Comment:
(non-blocking): Add godoc

```suggestion
// GetRoleBindingsForController returns the appropriate generator function for the given named controller that will reconcile role bindings in a namespace/project.
func GetRoleBindingsForController(controller string) projectRoleBindings {
```